lib/gc/ExecutionEngine/CPURuntime/Microkernel/BrgemmOnednn.cpp

```diff
@@ -66,20 +66,19 @@ static constexpr size_t SCRATCH_SIZE = 2 * 4096;
 // TODO(haixin): need to use custom thread management for scratch in the future?
 static thread_local char scratch[SCRATCH_SIZE] = {0};
 
-static std::unordered_map<int64_t, brgemm_cache_info_t> &get_tl_cache() {
-  thread_local std::unordered_map<int64_t, brgemm_cache_info_t> tl_cache;
+static std::vector<brgemm_cache_info_t> &get_tl_cache() {
+  thread_local std::vector<brgemm_cache_info_t> tl_cache;
   return tl_cache;
 }
+brgemm_desc_t desc;
```

Comment:
We are using a global temporary desc without syncronization?

Reply:
This is for save the desc object's address, which will be used in `g_cache.push_back(brgemm_cache_info_t{&desc, kernel, palette_buffer});`. Otherwise, the address will be invalid when `dispatch func` return.

Reply:
But now every `*desc` points to the same object, and the dispatch (will modify the global desc) has no synchronization at all?
If we really need to use pointer of desc, I think we can appoint an maximal amount of dispatched kernel, and reserve desc vector pool of that size to avoid vector resizing, then we can use pointer to vector element.